### PR TITLE
arpack-ng: Add variant for ICB option

### PR DESCRIPTION
**Pull request by sebastiangrimberg**
This has been in ARPACK-NG since v3.6.0.

### CODE REVIEW

File: var/spack/repos/builtin/packages/arpack-ng/package.py [hunk 1]
```diff
@@ -57,6 +57,7 @@ class ArpackNg(CMakePackage, AutotoolsPackage):
 
     variant("shared", default=True, description="Enables the build of shared libraries")
     variant("mpi", default=True, description="Activates MPI support")
+    variant("icb", default=False, description="Activates iso_c_binding support")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/arpack-ng/package.py]:
```suggestion
    variant("icb", default=False, when="@3.6:", description="Activates iso_c_binding support")
```

File: var/spack/repos/builtin/packages/arpack-ng/package.py [hunk 2]
```diff
@@ -85,6 +86,8 @@ class ArpackNg(CMakePackage, AutotoolsPackage):
         depends_on("libtool@2.4.2:", type="build")
         depends_on("pkgconfig", type="build")
 
+    conflicts("+icb", when="@:3.5.0")
+
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/arpack-ng/package.py]:
```suggestion
```

**Review by alalazo — CHANGES_REQUESTED** [review]
Can you check if using a conditional variant works too? It would be preferable, when possible, for clingo.

File: var/spack/repos/builtin/packages/arpack-ng/package.py (hunk 1 above)
Comment (sebastiangrimberg) [reply var/spack/repos/builtin/packages/arpack-ng/package.py]: Good catch, resolved in [9bcb31d](https://github.com/spack/spack/pull/34529/commits/9bcb31dfc61b8a203b4fe28121689f2360248566).

**Review by alalazo — APPROVED** [review]

**Pull request merged**